src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs

```diff
@@ -989,6 +989,10 @@ private PSModuleInfo ImportModule_LocallyViaName(ImportModuleOptions importModul
                 try
                 {
                     this.ArgumentList = new object[] { psSession };
+                    // since we already applied these filters remotely, we don't want to apply it locally where the generated module is always 1.0
```

Comment:
I am a little worried about changing these object properties here permanently.  It looks like this method is called within a loop of many kinds of different modules, including CIM and remote modules.  I think we should change and restore in the try/finally to be safe.

This is pretty confusing and I wonder if we can add a clearer comment?  Something like: "The correct module version has been imported from the remote session and created locally.  The locally created module always has a version of 1.0 regardless of the actual module version imported from the remote session, and version checking is no longer needed and will not work while importing this created local module".

I suppose the right thing to do is to ensure the generated module has the same version info as the module imported from the remote session.  But that is a much bigger change and can be handled in a separate issue.

Reply:
I'll update the comment with your suggestion.  

I initially looked to see about updating the proxy module version, but there were a couple of problems.  Not only was it not trivial to update the module version, based on my reading of the code, because you can export-pssession multiple modules, you actually don't have a single module version to use so from that perspective 1.0 makes sense.  Of course, we could special case single modules to have the same version number, but a bunch of plumbing code would be needed.

Reply:
As far as I could tell, it seemed safe to change it, but I can store it and change it back.